core/src/main/java/com/datastax/oss/driver/internal/core/loadbalancing/BasicLoadBalancingPolicy.java

```diff
@@ -422,4 +436,25 @@ protected NodeDistance computeNodeDistance(@NonNull Node node) {
   public void close() {
     // nothing to do
   }
+
+  // logs current list of
+  private void logLiveNodesByDc() {
+    // check trace level enabled just in case
+    if (!LOG.isTraceEnabled()) {
+      return;
+    }
+
+    LOG.trace(
+        "[{}] Current live nodes by dc: {{}}",
+        logPrefix,
+        liveNodes.dcs().stream()
+            .map(
+                dc ->
+                    dc
+                        + ": "
+                        + liveNodes.dc(dc).stream()
+                            .map(Objects::toString)
+                            .collect(Collectors.joining(", ")))
+            .collect(Collectors.joining(", ")));
```

Comment:
let me know if theres a saner way to format these than to put everything into a single line and let the maven plugin figure it out 😅